src/sidebar/components/excerpt.js

```diff
@@ -1,178 +1,189 @@
-    // annotation cards, this causes clicking on a quote to scroll the view to
-    // the selected annotation.
-    this.collapse = false;
-  };
-
-  this.showInlineControls = function() {
-    return this.overflowing && this.inlineControls;
-  };
-
-  this.bottomShadowStyles = function() {
-    return {
-      excerpt__shadow: true,
-      'excerpt__shadow--transparent': this.inlineControls,
-      'is-hidden': !this.isExpandable(),
-    };
-  };
-
-  // Test if the element or any of its parents have been hidden by
-  // an 'ng-show' directive
-  function isElementHidden() {
-    let el = $element[0];
-    while (el) {
-      if (el.classList.contains('ng-hide')) {
-        return true;
-      }
-      el = el.parentElement;
-    }
-    return false;
-  }
+const noop = () => {};
 
-  const overflowMonitor = new ExcerptOverflowMonitor(
-    {
-      getState: function() {
-        return {
-          animate: self.animate,
-          collapsedHeight: self.collapsedHeight,
-          collapse: self.collapse,
-          overflowHysteresis: self.overflowHysteresis,
-        };
-      },
-      contentHeight: function() {
-        const contentElem = $element[0].querySelector('.excerpt');
-        if (!contentElem) {
-          return null;
-        }
-        return contentElem.scrollHeight;
-      },
-      onOverflowChanged: function(overflowing) {
-        self.overflowing = overflowing;
-        if (self.onCollapsibleChanged) {
-          self.onCollapsibleChanged({ collapsible: overflowing });
-        }
-        // Even though this change happens outside the framework, we use
-        // $digest() rather than $apply() here to avoid a large number of full
-        // digest cycles if many excerpts update their overflow state at the
-        // same time. The onCollapsibleChanged() handler, if any, is
-        // responsible for triggering any necessary digests in parent scopes.
-        $scope.$digest();
-      },
-    },
-    window.requestAnimationFrame
+/**
+ * A container which truncates its content when they exceed a specified height.
+ *
+ * The collapsed state of the container can be handled either via internal
+ * controls (if `inlineControls` is `true`) or by the caller using the
+ * `collapse` prop.
+ */
+function Excerpt({
+  children,
+  collapse = false,
+  collapsedHeight,
+  inlineControls = true,
+  onCollapsibleChanged = noop,
+  onToggleCollapsed = noop,
+  overflowHysteresis = 0,
+  settings = {},
+}) {
+  const [collapsedByInlineControls, setCollapsedByInlineControls] = useState(
+    true
   );
 
-  this.contentStyle = overflowMonitor.contentStyle;
+  // Container for the excerpt's content.
+  const contentElement = useRef(null);
+
+  // Measured height of `contentElement` in pixels.
+  const [contentHeight, setContentHeight] = useState(0);
+
+  // Update the measured height of the content after the initial render and
+  // when the size of the content element changes.
+  const updateContentHeight = useCallback(() => {
```

Comment:
This may just be because the logic is what it is (and as simplified as it already can be), but it took me a while to grok the whole picture between here (line 80) and about line 96—the flow of state, control and what triggers what.

Part of this is simply that the `useCallback` hook is somewhat weirdly-named, and I always have to remind myself that's it's basically just memo-izing a fn reference.

I'm not sure I have any _suggestions_ here, it may be as straightforward as it can be already!